packages/teleport-shared/src/node-handlers/node-to-html/index.ts

```diff
@@ -22,7 +22,7 @@ const generateElementNode: NodeToHTML<UIDLElementNode, HastNode> = (node, params
     ? createHTMLNode(templateSyntax.customElementTagName(camelCaseToDashCase(elementType)))
     : createHTMLNode(elementType)
 
-  if (dependency) {
+  if (dependency && !templateSyntax.shouldOmitLocalDependencies) {
```

Comment:
you can use the same flag as the node-to-jsx, `dependencyHandling`: 'import' | 'ignore'. then add the condition in here